dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,147 @@
+import logging
+from datetime import datetime
+
+from analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+from analyzer_helper.discord.utils.is_user_bot import UserBotChecker
+
+
+class DiscordTransformRawData(TransformRawDataBase):
+
+    def __init__(self):
+        """
+        Initializes the class with a bot checker.
+        """
+        self.user_bot_checker = UserBotChecker()
+
+    def create_interaction(self, interaction_type, users_engaged_id, interaction_role):
+        return {
+            "name": interaction_type,
+            "users_engaged_id": users_engaged_id,
+            "type": interaction_role,
+        }
+
+    def create_receiver_interaction(self, data, interaction_type, engaged_user, author):
+        is_bot = self.user_bot_checker(author)
+        return {
+            "author_id": engaged_user,
```

Comment:
The `author_id` is the provided `author` in function parameters. the `engaged_user` would go into the line 38 when assigning the `users_engaged_id`.